app/services/post_alerter.rb

```diff
@@ -265,6 +269,22 @@ def category_watchers(topic)
       .pluck(:user_id)
   end
 
+  def tag_muters(topic)
+    topic
+      .tag_users
+      .includes(:user)
+      .notification_level_visible([TagUser.notification_levels[:muted]])
+      .map(&:user)
+  end
+
+  def category_muters(topic)
+    topic
+      .category_users
+      .includes(:user)
+      .where(notification_level: CategoryUser.notification_levels[:muted])
+      .map(&:user)
+  end
```

Comment:
I feel like there is a more efficient way to do this by doing it in a single query and ensuring the user is unique.